heap/heap.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <vector>
+
+class heap {
+    std::vector<int> tree;
+
+    void sift_up(int i);
+    void sift_down(int i);
+
+public:
+    heap(const std::vector<int>& array) : tree(array) {
+        // make heapify
+        for(int j=tree.size()/2; j>=0; j--){
+            sift_down(j);
+        }
+    }
```

Comment:
```suggestion
    heap(const std::vector<int>& array);
```
Оставь здесь только объявление, определение вынеси в .cpp файл по аналогии с sift_up, sift_down